plugins/source/hubspot/resources/services/crm/contacts_fetch.go

```diff
@@ -45,11 +51,11 @@ func fetchContacts(ctx context.Context, meta schema.ClientMeta, parent *schema.R
 			break
 		}
 		next := paging.GetNext()
-		after = next.After
-		if after == "" {
+		if next.After == "" {
 			break
 		}
+		after = next.After
 	}
 
-	return nil
+	return setCursor(ctx, cqClient, key, after)
```

Comment:
IMO, instead of `after` we should use the last value.

Example: we got the response in a single page (so `next.After` is empty). The next sync will resync all of the data.

Reply:
What do you mean? `after` stores the last pagination cursor. If there was no cursor available (because there is only one page) then there is no state to store and yes, some data will be resynced (but we're talking about less than one page of data). There is nothing we can do about that, though.